pkg/steps/secretrecordingclient/client.go

```diff
@@ -104,11 +104,20 @@ func (c *client) record(obj ctrlruntimeclient.Object) {
 }
 
 func (c *client) recordSecret(secret *v1.Secret) {
+	_, isServiceAccountSecret := secret.ObjectMeta.Annotations["kubernetes.io/service-account.name"]
 	var values []string
-	for _, value := range secret.Data {
+	for key, value := range secret.Data {
+		if key == "namespace" && isServiceAccountSecret {
+			// this is just never a useful thing to censor
```

Comment:
I don't want to do this, open to other ideas. We list all secrets in the namespace in order to figure out which we want to censor for the test pods, so we end up reading all of the values in this recording client even if we end up filtering the list of secrets after listing. Censoring the namespace ends up mangling e.g. the jUnit, etc.

Reply:
I would probably check on the type instead of the annotation to see if its a SA secret, but I don't really see a way around special casing the namespace

Reply:
Done